### PR TITLE
Handle exception when importing RetroGPTChunkDatasets

### PR DESCRIPTION
**Pull request by guyueh1**
# What does this PR do ?

Handle exception during import of RetroGPTChunkDatasets

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Handle Exception during import to allow a graceful fallback when importing RetroGPTChunkDatasets from megatron.

Currently on arm linux platform, `from megatron.core.datasets.retro.config import RetroGPTChunkDatasets` fails with an exception which is [a known issue](https://github.com/facebookresearch/faiss/issues/3936) with its dependency, faiss. When the exception happens, we want to gracefully fall back to the codepath without megatron retro support.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/nlp/data/language_modeling/megatron/retro_dataset.py
Comment (thomasdhc) [inline nemo/collections/nlp/data/language_modeling/megatron/retro_dataset.py:1]: LGTM

**Review by thomasdhc — APPROVED** [review]

**Comment by guyueh1** [comment]
@JimmyZhang12 could you trigger CICD again, let's get this merged

**Comment by guyueh1** [comment]
wait, need to resolve a confict

**Comment by guyueh1** [comment]
The bot pushed some changes to `.secrets.baseline` , not sure if that's normal; please start CICD @JimmyZhang12

**Comment by thomasdhc** [comment]
@ko3n1g Any idea what the secret baseline update is about?

**Comment by ko3n1g** [comment]
> @ko3n1g Any idea what the secret baseline update is about?

I have a bunch what’s going on. I’ll send a fix tmrw

**Comment by thomasdhc** [comment]
@guyueh1 @ko3n1g merged in the changes to fix the secret update. Let's merge main and revert that change. Should be able to merge after that. Thanks!

**Review by thomasdhc — APPROVED** [review]
lgtm

**Comment by guyueh1** [comment]
@thomasdhc CICD finished, looks good to merge

**Pull request merged**